plugins/alerting/pkg/alerting/drivers/alerting_driver_dynamic.go

```diff
@@ -194,9 +200,125 @@ func (a *AlertingManager) Update(ctx context.Context, conf *alertops.AlertingCon
 	if err != nil {
 		return nil, err
 	}
+	lg.Debug("editing statefulsets...")
+	// !! must edit statefulset pod annotations to trigger a SELF_DELETE from the
+	// !! mounted config symlink inside the alertmanager pod
+	controllerSvcData := &appsv1.StatefulSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      a.AlertingOptions.ControllerNodeService + "-internal",
+			Namespace: a.gatewayRef.Namespace,
+		},
+	}
+	err = a.k8sClient.Get(ctx, client.ObjectKeyFromObject(controllerSvcData), controllerSvcData)
+	if err != nil {
+		lg.Error(err)
+		return nil, err
+	}
+	numReplicas := controllerSvcData.Spec.Replicas
+	if numReplicas == nil {
+		lg.Error(err)
+		return nil, err
+	}
+	workerSvcData := &appsv1.StatefulSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      a.AlertingOptions.WorkerNodesService + "-internal",
+			Namespace: a.gatewayRef.Namespace,
+		},
+	}
+	var numWorkerReplicas *int32
+	zero := int32(0)
+	err = a.k8sClient.Get(ctx, client.ObjectKeyFromObject(workerSvcData), workerSvcData)
+	if k8serrors.IsNotFound(err) {
+		numWorkerReplicas = &zero
+	} else if numWorkerReplicas == nil {
+		numWorkerReplicas = &zero
+	} else {
+		numWorkerReplicas = workerSvcData.Spec.Replicas
+	}
```

Comment:
This logic doesn't look quite right - also the default value for Replicas is 1 if it's nil